lib/screens/v2/candidate_generator/widgets/elevator_closures.ex

```diff
@@ -0,0 +1,102 @@
+defmodule Screens.V2.CandidateGenerator.Widgets.ElevatorClosures do
+  @moduledoc false
+
+  alias Screens.Alerts.Alert
+  alias Screens.Config.Screen
+  alias Screens.Config.V2.{ElevatorStatus, PreFare}
+  alias Screens.RoutePatterns.RoutePattern
+  alias Screens.Stops.Stop
+  alias Screens.V2.WidgetInstance.ElevatorStatus, as: ElevatorStatusWidget
+
+  def elevator_status_instances(
+        %Screen{
+          app_params: %PreFare{
+            elevator_status: %ElevatorStatus{parent_station_id: parent_station_id}
+          }
+        } = config,
+        now
+      ) do
+    with {:ok, stop_sequences} <-
+           RoutePattern.fetch_stop_sequences_through_stop(parent_station_id),
+         {:ok, parent_station_map} <- Stop.fetch_parent_station_name_map(),
+         {:ok, elevator_closures, facility_id_to_name} <- fetch_elevator_closures(),
+         icon_map <- get_icon_map(elevator_closures) do
+      [
+        %ElevatorStatusWidget{
+          alerts: elevator_closures,
+          facility_id_to_name: facility_id_to_name,
+          stop_sequences: stop_sequences,
+          screen: config,
+          now: now,
+          station_id_to_name: parent_station_map,
+          station_id_to_icons: icon_map
+        }
+      ]
+    end
+  end
+
+  def fetch_elevator_closures do
+    case Screens.V3Api.get_json("alerts", %{
+           "filter[activity]" => "USING_WHEELCHAIR",
+           "include" => "facilities"
+         }) do
+      {:ok, result} ->
+        facilities =
+          result
+          |> get_in([
+            "included",
+            Access.filter(&(&1["type"] == "facility"))
+          ])
+          |> parse_facility_data()
+
+        elevator_closures =
+          result
+          |> Screens.Alerts.Parser.parse_result()
+          |> Enum.filter(fn
+            %Alert{effect: :elevator_closure} = alert -> alert
+            _ -> false
+          end)
+
+        {:ok, elevator_closures, facilities}
+
+      _ ->
+        :error
+    end
+  end
+
+  defp get_icon_map(elevator_closures) do
+    elevator_closures
+    |> get_parent_station_ids_from_entities()
+    |> Enum.map(fn station_id ->
+      {station_id, station_id |> Stop.create_station_with_routes_map() |> routes_to_icons()}
+    end)
+    |> Enum.into(%{})
+  end
+
+  defp get_parent_station_ids_from_entities(alerts) do
+    alerts
+    |> Enum.flat_map(fn %Alert{informed_entities: informed_entities} ->
+      informed_entities
+      |> Enum.map(fn %{stop: stop_id} -> stop_id end)
+      |> Enum.filter(&String.starts_with?(&1, "place-"))
+    end)
+  end
+
+  defp routes_to_icons(routes) do
+    routes
+    |> Enum.map(fn
+      %Screens.Routes.Route{type: :subway, id: id} -> id |> String.downcase() |> String.to_atom()
+      %Screens.Routes.Route{type: :light_rail} -> :green
```

Comment:
Not necessarily :green.  The Mattapan line is a light rail segment of the RL.  There is at least one station (Ashmont) on Mattapan that has an elevator

Reply:
Good to know. I added `id` to the pattern to only look at GL ids.

Reply:
Which of these conditionals will catch the Mattapan route?  Looks like it will fall to `%Screens.Routes.Route{type: type} -> type`, which will return the type `:light_rail` and doesn't map to a route icon

Reply:
Shouldn't it be an icon option for the widget?  Or, I guess it would be captured ok by the RL icon

Reply:
Good question. I will ask Mindy about this.

Reply:
Yep, Mattapan should have an icon! Went ahead and added it here. Mindy will work on an icon in the meantime, but that isn't needed for this.